src/components/WalletStatementModal/index.js

```diff
@@ -33,7 +52,13 @@ class WalletStatementModal extends React.Component {
                         width="100%"
                         seamless="seamless"
                         frameBorder="0"
-                        onLoad={() => this.setState({isLoading: false})}
+                        onLoad={() => {
+                            this.setState({isLoading: false});
+
+                            // We listen to a message sent from the iframe to the parent component when a link is clicked.
+                            // This lets us handle navigation in the app, outside of the iframe.
+                            window.onmessage = e => this.navigate(e.data);
```

Comment:
Should we verify the message is something we expect with a data type?  I'm worried we might intercept messages not intended to redirect and end up breaking redirect

Reply:
Nice! I added the data type check for url. Thanks @thienlnam!

Reply:
Ah sorry I should have clarified, I was thinking we should probably add another parameter on the event message so it would be like
{url: ____, type: 'STATEMENT_NAVIGATE'}

and then we can check

if (event.origin !== "statements.php" || !e.data || e.data.type !=  'STATEMENT_NAVIGATE')

That way we definitely won't listen pick up any other messages unintended for this

Reply:
Ahh that makes sense! I can add those. One thing that I noticed in my tests is that the `event.origin` only includes the domain and not the relative path, so checking for the `statements.php` origin might be tricky. I think checking for the even type might be enough though?

![Screen Shot 2022-03-16 at 4 28 26 PM](https://user-images.githubusercontent.com/22219519/158701970-2008c177-f6e7-4da7-88b2-69b9a59b112a.png)

Reply:
Yeah I agree that's fine, event type will be enough!